lighthouse-core/gather/driver.js

```diff
@@ -633,6 +661,7 @@ class Driver {
       })
       .then(_ => waitForLoad && this._waitForFullyLoaded(pauseAfterLoadMs,
           networkQuietThresholdMs, cpuQuietThresholdMs, maxWaitMs))
+      .then(_ => this._createIsolatedWorld())
```

Comment:
Would there be downsides to clearing on all navigations (as above) but calling `_createIsolatedWorld` lazily on next `evaluateAsync()` when `_isolatedExecutionContextId` isn't defined?

Reply:
done